retrieval/discovery/marathon/marathon_test.go

```diff
@@ -41,7 +41,7 @@ func TestMarathonSDHandleError(t *testing.T) {
 	var (
 		errTesting = errors.New("testing failure")
 		ch         = make(chan []*config.TargetGroup, 1)
-		client     = func(url string) (*AppList, error) { return nil, errTesting }
+		client     = func(url string, auth *config.BasicAuth) (*AppList, error) { return nil, errTesting }
```

Comment:
What's up with these method signature extensions? They are not used.
That said, the general feature would need a test.